### PR TITLE
feat: enforce trailing-comma style rule

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat: enforce trailing-comma style rule

## Linear Issue Reference

* LOG-196

## Testing

Tested by run the command `./gradlew checkCodeStyle`

### CODE REVIEW

File: build-logic/detekt/src/main/resources/detekt.yml [hunk 1]
```diff
@@ -6,6 +6,13 @@ style:
   ThrowsCount:
     active: false
 
+formatting:
+  active: true
+  TrailingComma:
+    active: true
+    allowTrailingComma: true
+    allowTrailingCommaOnCallSite: false
```
Comment (gao-sun) [inline build-logic/detekt/src/main/resources/detekt.yml:14]: should we enforce this as well since there's no transpiling issue in kotlin?

Comment (xiaoyijun) [reply build-logic/detekt/src/main/resources/detekt.yml:14]: it's optional in kotlin, so it's good to have a rule.

Comment (IceHe) [reply build-logic/detekt/src/main/resources/detekt.yml:14]: prefer always with commas

**Review by gao-sun — APPROVED** [review]

File: build-logic/detekt/src/main/resources/detekt.yml (hunk 1 above)
Comment (xiaoyijun) [reply build-logic/detekt/src/main/resources/detekt.yml]: updated

**Pull request merged**